### ISSUE TITLE
Cloning linode server with salt-cloud requires payment term

### ISSUE THREAD

**Issue by dnd** [saltstack/salt#30432]
I'm using 2015.8.3, and when trying to clone a node, I get the error:

```
[ERROR   ] Failed to create VM clone-test. Configuration value 'LinodeID' needs to be set
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/salt/cloud/__init__.py", line 1249, in create
    output = self.clouds[func](vm_)
  File "/usr/lib/python2.7/dist-packages/salt/cloud/clouds/linode.py", line 428, in create
    node_id = _clean_data(result)['LinodeID']
KeyError: 'LinodeID'
Error: There was a profile error: Failed to deploy VM
```

When I output the contents of the `result` dict from Linode, it contains the following:

```
{u'ACTION': u'linode.clone', u'DATA': {}, u'ERRORARRAY': [{u'ERRORCODE': 8, u'ERRORMESSAGE': u'Please select a valid payment term'}]}
```

I've never specified a payment term in my provider or profile configs when creating a machine, so I'm not sure what's going on here.

**Comment by rallytime** [comment]
@dnd Can you post your cloud provider and profile here, with any sensitive data stripped out?

**Comment by dnd** [comment]
```yaml
linode:
  driver: linode
  apikey: hunter2
  password: hunter2
  private_ip: True
  script: bootstrap-linode-networking.sh
  script_args: '-P git v2015.8.3'
  grains:
    provider: linode
```

**Comment by rallytime** [comment]
@dnd Thanks! Can you also post your profile information? I don't see any `size`, `image`, or `location` information with what you've posted here.

**Comment by dnd** [comment]
Here's the profile I'm using for the clone.

```yaml
clone_core_cache:
  provider: linode
  image: Ubuntu 14.04 LTS
  size: Linode 1024
  clonefrom: test-clone1
  location: Newark, NJ, USA
  script_args: -C
```

**Comment by rallytime** [comment]
@dnd It appears the cloning functionality is pretty broken in the 2015.8 branch right now. I'll be cleaning it up in the next couple of days. Thanks for posting your configs.

**Comment by rallytime** [comment]
Related to #30444

**Comment by rallytime** [comment]
@dnd I have fixed the cloning functionality in PR #30508. Some things to note after that fix goes in:
- In addition to specifying `-C` in the `script_args` argument, you'll also need to add `-F` to "force" the minion config file rewrite. Without the `-F`, the minion config file gets copied from the cloned VM, and the minion ID never gets reset. The new VM will have the same ID as the "to-be-cloned" VM.
- You should no longer need to specify the `location`, `size`, and `image` parameters in the `clone_core_cache` profile. You can still specify `location` and `size` if you want to, but if you want the cloned VMs to have the same datacenter location and size as the to-be-cloned VM, then you don't have to specify them again. (If you want them to be different, then you can specify `location` and `size`.) Specifying `image` has no effect, since the new VM will be a clone of the image specified in the `to-be-cloned` VM.

Basically all you'd need to get this to work again, after applying the changes in the above PR is:
```
clone_core_cache:
  provider: linode
  clonefrom: test-clone1
  script_args: -C -F
```

**Comment by cinterloper** [comment]
I get this same error, unrelated to cloning, when simply creating a Linode VM

```
[INFO    ] salt-cloud starting
[INFO    ] Creating Cloud VM salt-cloud
[ERROR   ] Failed to create VM salt-cloud. Configuration value 'LinodeID' needs to be set
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/salt/cloud/__init__.py", line 1274, in create
    output = self.clouds[func](vm_)
  File "/usr/lib/python2.7/dist-packages/salt/cloud/clouds/linode.py", line 458, in create
    node_id = _clean_data(result)['LinodeID']
KeyError: 'LinodeID'
[INFO    ] Creating Cloud VM edgegen
[ERROR   ] Failed to create VM edgegen. Configuration value 'LinodeID' needs to be set
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/salt/cloud/__init__.py", line 1274, in create
    output = self.clouds[func](vm_)
  File "/usr/lib/python2.7/dist-packages/salt/cloud/clouds/linode.py", line 458, in create
    node_id = _clean_data(result)['LinodeID']
KeyError: 'LinodeID'
```
this is in 2015.8.8, no 'cloneing' being used

```
linode_arch:
  provider: linode_public
  size: Linode 1024
  disk_size: 20000
  image: Arch Linux 2015.08
  deploy_command: curl -L https://bootstrap.saltstack.com | sudo sh -- -A '[REDACTED]'
  script_args: -A [REDACTED] git v2015.8.8
  minion:
    master: '[REDACTED]'
```

**Comment by rallytime** [comment]
@cinterloper Would you mind opening a new issue for the error you're seeing? I just tried my typical salt-cloud set up on 2015.8.8 with linode, and I am not seeing any errors, so we should handle your stacktrace/setup separately in a new issue. 

Be sure to provide your salt-cloud provider and profile configs (with sensitive info removed) in your new issue and we can take a closer look. Thanks!
